.github/workflows/ci.yml

```diff
@@ -0,0 +1,151 @@
+name: CI
+
+
+on:
```

Comment:
Does this mean these jobs will only run for master branch, PRs and tags? What about ordinary branches?

Reply:
changed to "all branches, tags starting w/ `v` and PRs (might come from other repos)".

Reply:
I'm a bit confused. maybe the UI is playing tricks on me but I still see

```yaml
on:
  push:
     branches:
       - master
```

plus some links to the explanation. My point is, this will disable builds for ordinary branches, doesn't it?

Reply:
Sorry, I had to revert the change. Yes. it will disable builds on ordinary branches, due to the reason illustrated in the links (you get double builds on PRs from this repo to this repo otherwise). If we wanna have release/backport branches in the future, we need to adapt this filter. I'm not entirely happy with this solution though, but having double builds on every BY PR seems worse.

Reply:
thb, this is kind of a blocker for me. I consider letting CI run for branches essential. My preferred choice would be to simply *not create PRs from this repo* but rather use a fork instead.

Reply:
Mmh, so there is a difference between `on.branch` and `on.pull_request`: The first runs on the tip of the branch, the latter on the PR merge result (see https://frontside.com/blog/2020-05-26-github-actions-pull_request/#how-does-pull_request-affect-actionscheckout ). If you rebase before merge, both are normally identical (or at least close enough). If the branch is somewhat stale, they are NOT identical and can also fail idependently. I think we now have the following requirements:

- running CI on non-PR branches (or before filing a --- potentially WIP --- PR)
- having the PR CI as close on the merge result is possible

With these two requirements, I don't see how we can workaround the doubled job issue. Would you be OK with that?

Side-note: this is also the normal behavior if a merge-train / merge-bot like [bors](https://bors.tech/) would be employed.

Reply:
I think I'm fine with this. This just makes the "problem" of testing too much more pressing. We desperately need to drop arrow versions...

I think that's actually a pretty cool functionality to get for free. It is annoying that it cannot be easily turned off, though.

Reply:
done